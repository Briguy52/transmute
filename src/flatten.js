import _flattenN from './internal/_flattenN';

/**
 * Flattens an iterable as deeply as possible.
 *
 * @example
 * // return List [ 1, 2, 3, 4, 5, 6 ]
 * flatten(List.of(List.of(1, List.of(2, 3)), List.of(4, 5, 6)));
 *
 * @param {Iterable} subject
 * @return {Iterable}
 */
export default function flatten(subject) {
  return _flattenN(Infinity, subject);
}
