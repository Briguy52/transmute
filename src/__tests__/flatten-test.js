import flatten from "../flatten";
import { List, Seq } from "immutable";

describe("transmute/flatten", () => {
  it("deeply flattens Lists", () => {
    expect(
      flatten(List.of(List.of(1, List.of(2, 3)), List.of(4, 5, 6)))
    ).toMatchSnapshot();
  });

  it("deeply flattens Seq", () => {
    expect(
      flatten(Seq.of(Seq.of(1, Seq.of(2, 3)), Seq.of(4, 5, 6)))
    ).toMatchSnapshot();
  });
});
