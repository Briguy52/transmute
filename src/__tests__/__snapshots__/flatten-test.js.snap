// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transmute/flatten deeply flattens Lists 1`] = `
Immutable.List [
  1,
  2,
  3,
  4,
  5,
  6,
]
`;

exports[`transmute/flatten deeply flattens Seq 1`] = `Immutable.Seq [â€¦]`;
